### ISSUE TITLE
Catching OpenSSL::SSL::SSLError ?

### ISSUE THREAD

**Issue by Nakilon** [sdsykes/fastimage#98]
There is a line of code that catches common network errors:
```ruby
Timeout::Error, SocketError, Errno::ECONNREFUSED, Errno::EHOSTUNREACH, Errno::ECONNRESET,
    ImageFetchFailure, Net::HTTPBadResponse, EOFError, Errno::ENOENT
```
How about to add `OpenSSL::SSL::SSLError` to it? I just faced it.  
`Zlib::BufError` can probably happen to.

**Comment by sdsykes** [comment]
Sure, we can do that

**Comment by sdsykes** [comment]
Fixed by 7bc37c590a122560dc20591dcec25eef9c91a2b6
